crates/bevy_tasks/src/single_threaded_task_pool.rs

```diff
@@ -1,12 +1,28 @@
-use alloc::{rc::Rc, sync::Arc};
+use alloc::{string::String, sync::Arc, vec::Vec};
 use core::{cell::RefCell, future::Future, marker::PhantomData, mem};
 
 use crate::Task;
 
+#[cfg(feature = "std")]
+use crate::executor::LocalExecutor;
+
+#[cfg(not(feature = "std"))]
+use crate::executor::Executor as LocalExecutor;
+
+#[cfg(feature = "std")]
 thread_local! {
-    static LOCAL_EXECUTOR: async_executor::LocalExecutor<'static> = const { async_executor::LocalExecutor::new() };
+    static LOCAL_EXECUTOR: LocalExecutor<'static> = const { LocalExecutor::new() };
 }
 
+#[cfg(not(feature = "std"))]
+static LOCAL_EXECUTOR: LocalExecutor<'static> = const { LocalExecutor::new() };
+
+#[cfg(feature = "std")]
+type ScopeResult<T> = alloc::rc::Rc<RefCell<Option<T>>>;
+
+#[cfg(not(feature = "std"))]
+type ScopeResult<T> = Arc<spin::Mutex<Option<T>>>;
```

Comment:
This isn't going to work with portable atomics feature, right?

Reply:
It _wasn't_! I've updated the PR to bring full support for `portable-atomic` and `critical-section`. With this PR as-is, you can compile and run on the Raspberry Pi Pico using:

```
cargo build -p bevy_tasks --target thumbv6m-none-eabi --no-default-features --features edge_executor,portable-atomic,critical-section
```

I had originally left this partially-implemented as I was going to circle back in a follow-up for this support. But looking at it, there's only a couple small changes (mostly in the `Cargo.toml`) needed to add the functionality anyway.